core/coreunix/add.go

```diff
@@ -67,42 +67,46 @@ type AddedObject struct {
 	Bytes int64  `json:",omitempty"`
 }
 
-func NewAdder(ctx context.Context, n *core.IpfsNode, out chan interface{}) (*Adder, error) {
-	mr, err := mfs.NewRoot(ctx, n.DAG, newDirNode(), nil)
+func NewAdder(ctx context.Context, p pin.Pinner, bs bstore.GCBlockstore, ds dag.DAGService) (*Adder, error) {
```

Comment:
Yes there was a case when I didn't have a node but everthing else. This was discussed with @whyrusleeping over IRC. 

There are checks in the code to check if fileadder.Out is nil and if so no attempt is written to the channel.